ThePhasesOfCraftship/2_best_practice_first/strategicDesignPart1/exercises/1_RefactoringTo4Tiers/begin/src/dto/ClassRequestDto.ts

```diff
@@ -0,0 +1,39 @@
+import { ErrorExceptionType } from "../constants"
+
+export type ClassRequestDto = {
+ name?: string
+ classId?: string
+ studentId?: string  
+}
+
+export default function createClassRequestDto(data: unknown) {
+  if(!data || typeof data !== 'object') {
+    throw new Error(ErrorExceptionType.ValidationError)
+  }
+  
+  const {name , classId, studentId} = data as Partial<ClassRequestDto>
```

Comment:
Since all keys on the dto are optional already, using Partial is redundant.
```suggestion
  const {name , classId, studentId} = data as ClassRequestDto
```